src/mavedb/routers/experiments.py

```diff
@@ -89,7 +89,13 @@ def search_experiments(search: ExperimentsSearch, db: Session = Depends(deps.get
     """
     Search experiments.
     """
-    return _search_experiments(db, None, search)
+    items = _search_experiments(db, None, search)
+    if items:
+        items = [
+            enrich_experiment_with_num_score_sets(exp, None)
+            for exp in items
+        ]
+    return items
```

Comment:
`_search_experiments` always returns a list, so we can just return this list comprehension directly. In the case where items is empty, the list comprehension will be empty and we will return an empty list (there's a good case to be made we should really return a 404 in that case). 
```suggestion
    return [
            enrich_experiment_with_num_score_sets(exp, None)
            for exp in items
        ]
```

Reply:
Thanks Ben! It also solves the incompatible type problems.